pkg/executor/operate_ddl_jobs.go

```diff
@@ -197,18 +199,37 @@ func (e *AlterDDLJobExec) processAlterDDLJobConfig(
 
 func (e *AlterDDLJobExec) updateReorgMeta(job *model.Job, byWho model.AdminCommandOperator) error {
 	for _, opt := range e.AlterOpts {
+		if opt.Value == nil {
+			continue
+		}
 		switch opt.Name {
 		case core.AlterDDLJobThread:
-			if opt.Value != nil {
-				cons := opt.Value.(*expression.Constant)
-				job.ReorgMeta.SetConcurrency(int(cons.Value.GetInt64()))
-			}
+			cons := opt.Value.(*expression.Constant)
+			job.ReorgMeta.SetConcurrency(int(cons.Value.GetInt64()))
 			job.AdminOperator = byWho
 		case core.AlterDDLJobBatchSize:
-			if opt.Value != nil {
-				cons := opt.Value.(*expression.Constant)
-				job.ReorgMeta.SetBatchSize(int(cons.Value.GetInt64()))
+			cons := opt.Value.(*expression.Constant)
+			job.ReorgMeta.SetBatchSize(int(cons.Value.GetInt64()))
+			job.AdminOperator = byWho
+		case core.AlterDDLJobMaxWriteSpeed:
+			var (
+				speed int64
+				err   error
+			)
+			v := opt.Value.(*expression.Constant)
+			switch v.RetType.EvalType() {
+			case types.ETString:
+				speedStr := opt.Value.(*expression.Constant).Value.GetString()
+				speed, err = units.RAMInBytes(speedStr)
+				if err != nil {
+					return errors.Trace(err)
+				}
+			case types.ETInt:
+				speed = opt.Value.(*expression.Constant).Value.GetInt64()
+			default:
+				return fmt.Errorf("the value %s for %s is invalid", opt.Name, opt.Value)
```

Comment:
this part is duplicate with what's in `planbuilder`, can we merge them

Reply:
Fix comments, PTAL again